pkg/runtime/reconciler.go

```diff
@@ -156,15 +156,15 @@ func (r *reconciler) sync(
 
 	latest, err := rm.ReadOne(ctx, desired)
 	if err != nil {
-		if latest != nil {
-			// this indicates, that even though ReadOne failed
-			// there is some changes available in the latest.RuntimeObject()
-			// (example: ko.Status.Conditions) which have been
-			// updated in the resource
-			// Thus, patchResource() call should be made here
-			_ = r.patchResource(ctx, desired, latest)
-		}
```

Comment:
I don't understand why this change was necessary. Why would ReadOne be returning err == NotFound && a non-nil `latest` variable?

Reply:
The earlier version of the code was having the err == NotFound && a non-nil latest scenario.
This PR fixes that.
With the changes in this PR; the resource would be patched when `err != NotFound && a not nil latest`. This scenario occurs when invalid input is given to SDK find.

Reply:
gotcha, thanks for the explanation. This is partly why I was a little hesitant about #574